sdk/go/common/resource/properties.go

```diff
@@ -129,6 +129,18 @@ func (ref ResourceReference) IDString() (value string, hasID bool) {
 	}
 }
 
+func (ref ResourceReference) Equal(other ResourceReference) bool {
+	if ref.URN != other.URN {
+		return false
+	}
+
+	vid, oid := ref.ID, other.ID
+	if vid.IsComputed() && oid.IsComputed() {
+		return true
```

Comment:
Why is this true? Couldn't the two refs be computed but be different once evaluated? I know this logic isn't new in this PR, just asking to clarify my understanding.

Also, in the previous version the condition was "or": `vid.IsComputed() || oid.IsComputed()` - is the change to "and" intentional?

Reply:
Yeh it's always struck me as information losing but if you want Equal to be a bool then you need to decided if unknowns are equal to all unknowns or no unknowns. Choosing the later gives you "the NaN" problem where `a != a`, so this was probably seen as easier to work with.